### ISSUE TITLE
Skip Accounts That Have Full Points

### ISSUE THREAD

**Issue by z0oinks** [xMNG/Microsoft-Rewards-Bot#9]
Hi!

Would it be possible to skip accounts that have full points for the day? 
I'm thinking if it sees you have full desktop search points, it goes directly to mobile search; if you have full mobile search points, it goes to quiz, etc.

This would be nice in case you add an account after you've run the script, or if you wanna run it a second time to make sure everything just done.

**Comment by xMNG** [comment]
Hi there, the script should already do this, providing that MS recognizes that you are logged in and shows the points page. What does your log file look like?

**Comment by ShoGinn** [comment]
@xMNG they are right, If you run the script now with al searches done it does the 20/30 searches before checking if it needs to.

**Comment by xMNG** [comment]
Hm the code should enter the first search term and then check to see if the points are completed for that category.

`        for num, item in search_terms:
            try:
                # clears search bar and enters in next search term
                time.sleep(1)
                wait_until_visible(By.ID, 'sb_form_q', 30)
                clear_by_id('sb_form_q')
                send_key_by_id('sb_form_q', item)
                time.sleep(0.1)
                send_key_by_id('sb_form_q', Keys.RETURN)
                # prints search term and item, limited to 80 chars
                logging.debug(msg=f'Search #{num}: {item[:80]}')
                time.sleep(random.randint(3, 4))  # random sleep for more human-like, and let ms reward website keep up.

                # check to see if search is complete, if yes, break out of loop
                if num % search_limit == 0:
                    if mobile_search:
                        # in mobile mode, get point total does not work if no search is done, URL = 404
                        if get_point_total(mobile=True):
                            logging.info(msg=f'Stopped at search number {num}')
                            return
                        # if point total not met, return to search page
                        browser.get(BING_SEARCH_URL)
                    else:
                        if get_point_total(pc=True):
                            logging.info(msg=f'Stopped at search number {num}')
                            return
browser.get(BING_SEARCH_URL)`

The code should only finish 20 or 30 search terms if the chrome ext url is not loading properly. I will adjust the code to check for the url first.

**Comment by ShoGinn** [comment]
@xMNG true, it should. But unfortunately due to the way chrome handles the chrome extension. It never finds the points in mobile.

there is a way for chromedriver to scroll to an object using:

location_once_scrolled_into_view

https://seleniumhq.github.io/selenium/docs/api/py/webdriver_remote/selenium.webdriver.remote.webelement.html?highlight=location_once_scrolled#selenium.webdriver.remote.webelement.WebElement.location_once_scrolled_into_view

In your get_point_total function I'd modify it like this:

```
def get_point_total(pc=False, mobile=False, log=False):
    """
    Checks for points for pc/edge and mobile, logs if flag is set
    :return: Boolean for either pc/edge or mobile points met
    """
    browser.get(POINT_TOTAL_URL)
    # get number of total number of points
    wait_until_visible(By.XPATH, '//*[@id="flyoutContent"]', 30)
    pcsearch = browser.find_element_by_class_name('pcsearch')
    pcsearch.location_once_scrolled_into_view
    try:
        current_point_total = list(map(int, browser.find_element_by_class_name('credits2').text.split(' of ')))[0]
        # get pc points
        current_pc_points, max_pc_points = map(int, browser.find_element_by_class_name('pcsearch').text.split('/'))
        # get mobile points
        current_mobile_points, max_mobile_points = \
            map(int, browser.find_element_by_class_name('mobilesearch').text.split('/'))
        # get edge points
        current_edge_points, max_edge_points = map(int,
                                                   browser.find_element_by_class_name('edgesearch').text.split('/'))
    except ValueError as e:
        return False

    # if log flag is provided, log the point totals
    if log:
        logging.info(msg=f'Total points = {current_point_total}')
        logging.info(msg=f'PC points = {current_pc_points}/{max_pc_points}')
        logging.info(msg=f'Edge points = {current_edge_points}/{max_edge_points}')
        logging.info(msg=f'Mobile points = {current_mobile_points}/{max_mobile_points}')

    # if pc flag, check if pc and edge points met
    if pc:
        if current_pc_points < max_pc_points or current_edge_points < max_edge_points:
            return False
        return True
    # if mobile flag, check if mobile points met
    if mobile:
        if current_mobile_points < max_mobile_points:
            return False
        return True
```

**Comment by ShoGinn** [comment]
This should be fixed with #13

**Comment by MacMarde** [comment]
Hi I am new here. I really appreciate your bot and your work.

I am trying to use the bot with my German accounts. I have set the website language to US English and I think the website does now look like with a real US account.

But I am also having the problem that the bot doesnt recognize accounts with full points. (Most probably the bot also does not work corrctly with empty points. But as for today all my accounts have full points)
I just started an account with full points. I can see in the log that the bot made 152 mobile searches, then started a new HTTP connection for making the PC searches. It seems like for every PC search I am now getting the following console error: 
"ERROR:web_contents_delegate.cc(197)] WebContentsDelegate::CheckMediaAccessPermission: Not supported. "

I am willing to contribute to the project as much as I can. Please help me to make it work with my german accounts. I do know a little bit about coding. Maybe we can work together to implement support for german accounts. 

Now Im also got this console error for some PC searches (I started from the Command prompt with administartor rights):
[0504/075749.009:INFO:CONSOLE(57)] "Uncaught SecurityError: Failed to set the 'href' property on 'Location': The current window does not have permission to navigate the target frame to 'https://login.microsoftonline.com/common/oauth2/authorize?client_id=9ea1ad79-fdb6-4f9a-8bc3-2b70f96e34c7&response_type=id_token&nonce=ecf2dfbb-7174-43f0-a7d5-4ebea31a4cbc&redirect_uri=https%3a%2f%2fwww.bing.com%2forgid%2fidtoken%2fnosignin&scope=openid&response_mode=form_post&msafed=0&prompt=none&iframe-request-id=4c4e5830-6d49-4acf-9b3c-58e561180900'.", source: https://login.microsoftonline.com/common/oauth2/authorize?client_id=9ea1ad79-fdb6-4f9a-8bc3-2b70f96e34c7&response_type=id_token&nonce=ecf2dfbb-7174-43f0-a7d5-4ebea31a4cbc&redirect_uri=https%3a%2f%2fwww.bing.com%2forgid%2fidtoken%2fnosignin&scope=openid&response_mode=form_post&msafed=0&prompt=none (57)"

It made over 100 PC searches so far. I think Im gonna stop it now.
